pretyping/tacred.ml

```diff
@@ -322,64 +341,125 @@ let deactivate_delta allowed_reds =
   (* Act both on Delta and transparent state as not all reduction functions work the same *)
   RedFlags.(red_add_transparent (red_sub allowed_reds fDELTA) TransparentState.empty)
 
-(* [compute_consteval] stepwise expands an arbitrary long sequence of
+(* [compute_constant_elimination] stepwise expands an arbitrary long sequence of
     reversible constants, eventually refolding to the initial constant
     for unary fixpoints and to the last constant encapsulating the Fix
     for mutual fixpoints *)
 
-let compute_consteval ((cache,_),allowed_reds as cache_reds) env sigma ref u =
```

Comment:
I think we should introduce a proper record for reduction caches.

Reply:
Eventually yes. I could do it as soon as this PR if you insist, or, maybe, simpler to me, at the time of rebasing PR #18580 which also adds a new component.